cloudferrylib/os/actions/vm_snapshot.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+
+from cloudferrylib.base.action import action
+from cloudferrylib.utils import utils
+
+
+LOG = utils.get_log(__name__)
+VM_STATUSES = "VM_STATUSES"
+
+
+class VmSnapshotBasic(action.Action):
+
+    """Base class that contains common to vm_snapshot methods"""
+
+    def get_compute_resource(self):
+        """returns cloudferry compute resource"""
+        return self.cloud.resources[utils.COMPUTE_RESOURCE]
+
+    @property
+    def namespace_variable(self):
+        """returns unique for cloud variable from namespace"""
+        return '_'.join([VM_STATUSES, self.cloud.position])
+
+    def get_list_of_vms(self):
```

Comment:
Why we do it instead of reuse of `compute.read_info()`? It is already made there.